splunk_add_on_ucc_framework/dashboard.py

```diff
@@ -136,6 +142,26 @@ def generate_dashboard(
     else:
         panels = global_config.dashboard.get("panels", [])
         panel_names = [panel["name"] for panel in panels]
-        content = generate_dashboard_content(addon_name, panel_names)
+        custom_components = ""
+        if PANEL_CUSTOM in panel_names:
+            try:
+                dashboard_components_path = os.path.abspath(
+                    os.path.join(
+                        global_config.original_path,
+                        os.pardir,
+                        "dashboard_components.txt",
+                    )
+                )
+                with open(dashboard_components_path) as file:
+                    custom_components = "".join(file.readlines())
+            except FileNotFoundError:
+                sys.exit(
+                    "custom dashboard page set but dashboard_components.txt file not found"
+                )
+            if not custom_components:
+                sys.exit(
+                    "custom dashboard page set but dashboard_components.txt file is empty"
+                )
+        content = generate_dashboard_content(addon_name, panel_names, custom_components)
```

Comment:
Let's add a XML validation step here just to make sure that resulting XML is actually a valid XML.

Reply:
You can use `defusedxml`.

Reply:
https://pypi.org/project/defusedxml/